File: src/test/java/com/ject/studytrip/stamp/fixture/CreateStampRequestFixture.java
package com.ject.studytrip.stamp.fixture;

import com.ject.studytrip.stamp.presentation.dto.request.CreateStampRequest;
import java.time.LocalDate;

public class CreateStampRequestFixture {

    private String name = "TEST STAMP";
    private java.time.LocalDate endDate = java.time.LocalDate.now().plusDays(7);

    public CreateStampRequestFixture withName(String name) {
        this.name = name;
        return this;
    }

    public CreateStampRequestFixture withEndDateInPast() {
        this.endDate = LocalDate.now().minusDays(1);
        return this;
    }

    public CreateStampRequestFixture withEndDateAfterTripEndDate() {
        this.endDate = LocalDate.now().plusDays(100);
        return this;
    }

    public CreateStampRequest build() {
        return new CreateStampRequest(name, endDate);
    }
}

File: src/test/kotlin/com/ject/studytrip/stamp/fixture/CreateStampRequestFixture.kt
package com.ject.studytrip.stamp.fixture

import com.ject.studytrip.stamp.presentation.dto.request.CreateStampRequest
import java.time.LocalDate

class CreateStampRequestFixture(
    private val name: String = "TEST 스탬프 이름",
    private val endDate: LocalDate? = LocalDate.now().plusDays(7),
) {
    fun withName(name: String): CreateStampRequestFixture = CreateStampRequestFixture(name, endDate)

    fun withEndDate(endDate: LocalDate?): CreateStampRequestFixture = CreateStampRequestFixture(name, endDate)

    fun withEndDateAfterTripEndDate(): CreateStampRequestFixture = CreateStampRequestFixture(name, LocalDate.now().plusDays(100))

    fun build() = CreateStampRequest(name, endDate)
}
